### PR TITLE
Minetest bug fix

### PR DESCRIPTION
**Pull request by johnlockard**
This update fixes version 5.1.0 main.cpp missing header file for irrlicht.

### CODE REVIEW

**Comment by johnlockard** [comment]
I think this is the error.

======
2019-11-03 08:17:46.252 GMT [51641] DETAIL:  Failed system call was shmget(key=5432005, size=56, 03600).
2019-11-03 08:17:46.252 GMT [51641] HINT:  This error does *not* mean that you have run out of disk space.  It occurs either if all available shared memory IDs have been taken, in which case you need to raise the SHMMNI parameter in your kernel, or because the system's overall limit for shared memory has been reached.
====

But I don't know how to raise the shared memory IDs. Does anyone know how to do this?

File: Formula/minetest.rb [hunk 1]
```diff
@@ -3,12 +3,12 @@ class Minetest < Formula
   homepage "https://www.minetest.net/"
 
   stable do
-    url "https://github.com/minetest/minetest/archive/5.0.1.tar.gz"
-    sha256 "aa771cf178ad1b436d5723e5d6dd24e42b5d56f1cfe9c930f6426b7f24bb1635"
+    url "https://github.com/johnlockard/minetest/archive/5.1.0.tar.gz"
```
Comment (Bo98) [inline Formula/minetest.rb]: This doesn't look official. Why aren't you using the official repository?

If there's a particular patch that's required, you can do that separately in a `patch do` block, assuming the patch has been accepted upstream:

```ruby
patch do
  url "https://github.com/org/repo/commit/whatever.patch?full_index=1"
  sha256 "abcdef"
end
```

Very similar to the `resource do` block.

File: Formula/minetest.rb [hunk 2]
```diff
@@ -29,11 +29,16 @@ class Minetest < Formula
...
+  depends_on "hiredis"
   depends_on "irrlicht"
   depends_on "jpeg"
+  depends_on "leveldb"
   depends_on "libogg"
   depends_on "libvorbis"
   depends_on "luajit"
+  depends_on "postgresql"
+  depends_on "spatialindex"
+  depends_on :x11
```
Comment (Bo98) [inline Formula/minetest.rb]: Is this needed? We generally avoid X11 where possible.

File: Formula/minetest.rb [hunk 3]
```diff
@@ -29,11 +29,16 @@ class Minetest < Formula
...
   depends_on "gettext"
+  depends_on "hiredis"
   depends_on "irrlicht"
   depends_on "jpeg"
+  depends_on "leveldb"
   depends_on "libogg"
   depends_on "libvorbis"
   depends_on "luajit"
+  depends_on "postgresql"
+  depends_on "spatialindex"
```
Comment (Bo98) [inline Formula/minetest.rb]: Can you explain a bit behind the extra dependencies. Are they optional? If so, what benefit do each bring?

File: Formula/minetest.rb [hunk 4]
```diff
@@ -59,4 +64,18 @@ def caveats
...
+    # --help and --version produce output directly.
+    # --speedtests and --videomodes need user data directory in order to work.
+    # --info and --trace need user data directory and run the game.
+    #
+    # --run-unittests and --speedtests do not work for Homebrew.
+    #
+    # Debug File: all test information should wind up in here.
+    #
+    (testpath/"Library/Application Support/minetest/debug.txt").write("")
+    system "#{prefix}/minetest.app/Contents/MacOS/minetest", "--videomodes"
```
Comment (Bo98) [inline Formula/minetest.rb]: Considering this uses a `.app`, I think it is better for this formula to be moved to Homebrew Cask.

https://docs.brew.sh/Acceptable-Formulae#stuff-that-builds-an-app

**Review by Bo98 — CHANGES_REQUESTED** [review]
The error you mentioned should now be fixed - apologies for the disruption. I can restart the CI, but there's a few concerns I have first.

**Comment by fxcoudert** [comment]
Closing due to lack of response.

**Pull request closed without merging**